mutable_batch/src/payload/partition.rs

```diff
@@ -846,6 +1082,28 @@ mod tests {
         }
     }
 
+    prop_compose! {
+        /// Yield a Vec containing an identical timestamp run of random length,
+        /// up to `max_run_len`,
+        fn arbitrary_timestamp_run(max_run_len: usize)(v in 0_i64..i64::MAX, run_len in 1..max_run_len) -> Vec<i64> {
+            let mut x = Vec::with_capacity(run_len);
+            x.resize(max_run_len, v);
+            x
+        }
+    }
+
+    /// Yield a Vec of timestamp values that more accurately model real
+    /// timestamps than pure random selection.
+    ///
+    /// Runs of identical timestamps are generated with
+    /// [`arbitrary_timestamp_run()`], which are then shuffled to produce a list
+    /// of timestamps with limited repeats, sometimes consecutively.
+    fn arbitrary_timestamps() -> impl Strategy<Value = Vec<i64>> {
+        proptest::collection::vec(arbitrary_timestamp_run(6), 10..100)
```

Comment:
I verified that this test does indeed fail now when the fix commit is reverted! 🎉 

However, the "minimal failing input" that proptest came back with had 60 points, because of this `10` here. I don't think it actually requires 60 points to trigger this bug (I bet it could be done with 3), so I tried changing this:

```suggestion
        proptest::collection::vec(arbitrary_timestamp_run(6), 0..100)
```

and now I'm getting "minimal failing input" of 18 points (I'm guessing now we're hitting the limit of how many times proptest tries to minimize the input? not sure). I'd rather get a test case of 18 than 60, is there a reason 10 is important here?

Reply:
The intent is to drive a meaningful number of rows through the partitioner in tests - it shouldn't affect shrinking, but you already figured that out!